---
/steinicp/SteinMICP:
   ros__parameters:
      class_type: "SteinMICP"
      SteinICP_parameters:
           optimizer : "Adam"   # only used for original Stein ICP
           iterations: 70
           batch_size: 200
           particle_size: 40
           normalize: false
           lr: 0.03
           max_dist: 1.0
           frame_gap_seconds: 0.05
           using_EarlyStop: true
           convergence_steps: 3
           convergence_threshold: 0.0001
           deskew_cloud: true
           max_range: 100.0
           min_range: 5.0       # 0.3 for New college cloister, 5.0 for others
           voxel_size: 1.0
           map_voxel_max_points: 20
           map_voxel_size: 1.0
           USE_Segmentation: false

           use_weight_mean: true

           save_particles: true
           pub_cloud: true

      lidar_param:
           topic: "/kitti/velo/pointcloud"

      imu_param:
           topic: "/kitti/oxts/imu_unsynced"

           rot_rw: [ 20., 20., 20. ]          # [deg/sqrt(hr)]
           vel_rw: [ 200., 200., 200. ]          # [m/s/sqrt(hr)]
           gbstd: [ 20.0, 20.0, 20.0 ]        # [deg/hr]
           abstd: [ 500.0, 500.0, 500.0 ]     # [mGal]
           corrtime: 1.0                    # [hr]

      lio_param:
           initposstd: [ 0.05, 0.05, 0.05 ]
           # initial velocity std, north, east and down speed in n-frame. [m/s, m/s, m/s]
           initvelstd: [ 0.05, 0.05, 0.05 ]
           # initial attitude std, roll, pitch and yaw std. [deg, deg, deg]
           initattstd: [ 0.1, 0.1, 0.1 ]

           # sensor calibration parameter transformation from imu frame to lidar frame
           extrinsic_t: [ 0.81, -0.32, 0.8 ]
           extrinsic_R: [ 1., 0., 0.,
                          0., 1., 0.,
                          0., 0., 1. ]




